### PR TITLE
Applied changes in the API to make it universal

### PR DESCRIPTION
**Pull request by ethouris**
These changes embrace:

1. Making the `srt_send` and `srt_sendmsg` (plus `srt_sendmsg2`) do the same thing internally. What exactly implementation is going to be done, is dependent on the `SRTO_MESSAGEAPI` flag. These functions effectively differ by the parameter set.
2. The `srt_recv` and `srt_recvmsg` are provided separately only for convenience any symmetry, they do exactly the same thing.
3. The `...msg2` API uses an extended information in the `SRT_MSGCTRL` structure for a case when an extended information is needed. This may be extended in future with extra information, upon necessity, without further changes in the API.
4. Added file-only functions `srt_sendfile` and `srt_recvfile`, and example apps `sendfile` and `recvfile`. They are ports of the same apps derived from UDT and they do exactly the same thing.

### CODE REVIEW

File: srtcore/core.cpp [hunk 1]
```diff
@@ -4678,24 +4779,70 @@ int CUDT::sendmsg(const char* data, int len, int msttl, bool inorder, uint64_t s
...
+        throw CUDTException(MJ_NOTSUP, MN_INVAL, 0);
+    }
+
+    if (m_bMessageAPI)
+        return receiveMessage(data, len, mctrl);
+
+    return receiveBuffer(data, len);
+}
 
+int CUDT::receiveMessage(char* data, int len, SRT_MSGCTRL* mctrl)
```
Comment (rndi) [inline srtcore/core.cpp]: There ought be a debug assert down this function checking for msgctl != NULL. Alternative is to change msgctl to be a reference.

File: srtcore/core.cpp [hunk 2]
```diff
@@ -4899,22 +5047,37 @@ int64_t CUDT::sendfile(fstream& ifs, int64_t& offset, int64_t size, int block)
...
    }
 
-   int64_t tosend = size;
-   int unitsize;
-
    // positioning...
    try
    {
-      ifs.seekg((streamoff)offset);
+       if (size == -1)
```
Comment (rndi) [inline srtcore/core.cpp]: ident != 3 spaces

File: srtcore/srt.h [hunk 3]
```diff
@@ -514,16 +519,63 @@ SRT_API extern int srt_getsockopt(SRTSOCKET u, int level /*ignored*/, SRT_SOCKOP
...
+// removed.
+
+// Sending
+SRT_API extern int srt_send(SRTSOCKET u, const char* buf, int len, ...);
 SRT_API extern int srt_sendmsg(SRTSOCKET u, const char* buf, int len, int ttl/* = -1*/, int inorder/* = false*/);
-SRT_API extern int srt_recvmsg(SRTSOCKET u, char* buf, int len);
 SRT_API extern int srt_sendmsg2(SRTSOCKET u, const char* buf, int len, SRT_MSGCTRL *mctrl);
+
+// Receiving
+SRT_API extern int srt_recv(SRTSOCKET u, char* buf, int len, ...);
```
Comment (rndi) [inline srtcore/srt.h]: Please avoid variadics if at all possible

File: srtcore/srt_c_api.cpp [hunk 4]
```diff
@@ -94,31 +116,64 @@ int srt_getsockflag(SRTSOCKET u, SRT_SOCKOPT opt, void* optval, int* optlen)
 int srt_setsockflag(SRTSOCKET u, SRT_SOCKOPT opt, const void* optval, int optlen)
 { return CUDT::setsockopt(u, 0, opt, optval, optlen); }
 
-int srt_send(SRTSOCKET u, const char * buf, int len, int flags) { return CUDT::send(u, buf, len, flags); }
-int srt_recv(SRTSOCKET u, char * buf, int len, int flags) { return CUDT::recv(u, buf, len, flags); }
+int srt_send(SRTSOCKET u, const char * buf, int len, ...) { return CUDT::send(u, buf, len, 0); }
+int srt_recv(SRTSOCKET u, char * buf, int len, ...) { return CUDT::recv(u, buf, len, 0); }
```
Comment (rndi) [inline srtcore/srt_c_api.cpp]: Please avoid variadics if at all possible

File: srtcore/core.cpp (hunk 2 above)
Comment (ethouris) [reply srtcore/core.cpp]: Whole functions have been reformatted to 4-spaced indents. We can discuss total reformatting of the code as a separate activity; the 3-spaced indents come from the original UDT and it's a little bit clumsy. No one really does that in the OS world and 4-spaced indents are rather generally preferred.

**Comment by rndi** [comment]
Thank you @ethouris !

**Pull request merged**